report-generator/src/main/java/com/teamscale/report/jacoco/JaCoCoXmlReportGenerator.java

```diff
@@ -53,11 +52,36 @@ public JaCoCoXmlReportGenerator(List<File> codeDirectoriesOrArchives,
 		this.logger = logger;
 	}
 
-	/** Creates the report. */
-	public String convert(Dump dump) throws IOException {
-		ByteArrayOutputStream output = new ByteArrayOutputStream();
+
+	/**
+	 * Creates the report and writes it to a file.
+	 *
+	 * @return The file object of for the converted report or null if it could not be created
+	 */
+	public File convert(Dump dump, String filePath) throws IOException {
+		File outputFile = createOutputFile(filePath);
+		if (outputFile == null) {
+			return null;
+		}
+		FileOutputStream output = new FileOutputStream(outputFile);
 		convertToReport(output, dump);
-		return output.toString(FileSystemUtils.UTF8_ENCODING);
+		return outputFile;
+	}
+
+	private File createOutputFile(String filePath) {
+		File outputFile = new File(filePath);
+
+		try {
+			boolean fileCreated = outputFile.createNewFile();
```

Comment:
please use java NIO where possible as it uses exceptions instead of boolean return codes, so we are forced to check the error cases and can't forget accidentally. In this case: `Files.createFile`.  also, in this case, you don't log anything which is bad for the user since they don't know that anything went wrong but don't see any coverage either. So they have no way of knowing what the problem is and thus no way of fixing it themselves